src/Http/WebUtilities/test/FormPipeReaderTests.cs

```diff
@@ -284,6 +284,32 @@ public void TryParseFormValues_MultiSegmentSplitAcrossSegmentsThatNeedDecodingWo
             Assert.Equal("wow", dict["\"%-.<>\\^_`{|}"]);
         }
 
+        [Fact]
+        public void TryParseFormValues_MultiSegmentExceedKeyLengthThrows()
+        {
+            var readOnlySequence = ReadOnlySequenceFactory.CreateSegments(Encoding.UTF8.GetBytes("foo=bar&baz=bo"), Encoding.UTF8.GetBytes("o&t="));
```

Comment:
This fails on the first key in the first sequence, it never gets to the split value (which should be a split key for this test).

Reply:
Do you need a separate test for a trailing key that exceeds the limit?